api/proto/teleport/legacy/types/types.proto

```diff
@@ -6213,6 +6215,125 @@ enum PluginStatusCode {
   SLACK_NOT_IN_CHANNEL = 10;
 }
 
+// PluginOktaStatusDetailsV1 contains the details for the running Okta plugin.
+message PluginOktaStatusDetailsV1 {
+  // SSODetails are status details relating to SSO.
+  PluginOktaStatusDetailsSSO sso_details = 2;
+
+  // AppGroupSyncDetails are status details relating to synchronizing apps and
+  // groups from Okta.
+  PluginOktaStatusDetailsAppGroupSync app_group_sync_details = 3;
+
+  // UsersSyncDetails are status details relating to synchronizing users from
+  // Okta.
+  PluginOktaStatusDetailsUsersSync users_sync_details = 4;
+
+  // ScimDetails are status details relating to SCIM integration with
+  // Okta.
+  PluginOktaStatusDetailsSCIM scim_details = 5;
+
+  // AccessListSyncDetails are status details relating to synchronizing access
+  // lists from Okta.
+  PluginOktaStatusDetailsAccessListsSync access_lists_sync_details = 6;
+}
+
+// PluginOktaStatusDetailsSSO are details related to the
+// current status of the Okta integration w/r/t SSO.
+message PluginOktaStatusDetailsSSO {
+  bool enabled = 1;
+  string app_id = 2;
+  string app_name = 3;
+}
+
+// PluginOktaStatusDetailsAppGroupSync are details related to the
+// current status of the Okta integration w/r/t application and group
+// sync.
+message PluginOktaStatusDetailsAppGroupSync {
+  // LastSuccessful is the date of the last successful run.
+  google.protobuf.Timestamp last_successful = 1 [
+    (gogoproto.stdtime) = true,
+    (gogoproto.nullable) = true,
+    (gogoproto.jsontag) = "last_successful"
+  ];
+
+  // LastFailed is the date of the last failed run.
+  google.protobuf.Timestamp last_failed = 2 [
+    (gogoproto.stdtime) = true,
+    (gogoproto.nullable) = true,
+    (gogoproto.jsontag) = "last_failed"
+  ];
+
+  // NumAppsSynced is the total number of apps synchronized.
+  int32 num_apps_synced = 3;
+
+  // NumAppsSynced is the total number of groups synchronized.
+  int32 num_groups_synced = 4;
+}
```

Comment:
Do we need sync status "string" field? So if there is an error, we can a write verbose error message to the status instead of just forwarding the PluginStatusCode?

Reply:
That's not a bad idea. Adding.